ddtrace/llmobs/_llmobs.py

```diff
@@ -1351,13 +1365,18 @@ def submit_evaluation(
     def _inject_llmobs_context(cls, span_context: Context, request_headers: Dict[str, str]) -> None:
         if cls.enabled is False:
             return
-        active_context = cls._instance._current_trace_context()
-        if active_context is None:
-            parent_id = ROOT_PARENT_ID
-        else:
-            parent_id = str(active_context.span_id)
+
+        active_span = cls._instance._llmobs_context_provider.active()
+        active_context = active_span.context if isinstance(active_span, Span) else active_span
+
+        parent_id = str(active_context.span_id) if active_context is not None else ROOT_PARENT_ID
+        ml_app = active_span._get_ctx_item(ML_APP) if isinstance(active_span, Span) else config._llmobs_ml_app
```

Comment:
for distributed tracing across multiple services we have to be sure that `active_span._get_ctx_item(ML_APP)` is the propagated service. Ensure that at the point this is called that the propagated service is set on the span. If it happens on span finish it might be too late